File: mvn-advent-of-code/src/test/java/advent/twenty_eighteen/Day24ImmuneSystemSimulator20XXTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day24ImmuneSystemSimulator20XXTest {
    @Test
    public void checkReal() {
        Day24ImmuneSystemSimulator20XX im = new Day24ImmuneSystemSimulator20XX("/twenty_eighteen/Day24-ImmuneSystemSimulator20XX-input.txt");

        assertEquals(22859, im.getPart1Answer().intValue());
        assertEquals(2834, im.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        Day24ImmuneSystemSimulator20XX im = new Day24ImmuneSystemSimulator20XX("/twenty_eighteen/Day24-ImmuneSystemSimulator20XX-example#1.txt");

        assertEquals(5216, im.getPart1Answer().intValue());
        assertEquals(51, im.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_eighteen/Day24ImmuneSystemSimulator20XXTest.scala
package advent.twenty_eighteen

import org.scalatest.{FlatSpec, Matchers}

class Day24ImmuneSystemSimulator20XXTest  extends FlatSpec with Matchers {
  "2018 Day 24 - Input" should "calculate answers" in {
    val dp = new Day24ImmuneSystemSimulator20XX("/twenty_eighteen/Day24-ImmuneSystemSimulator20XX-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day24ImmuneSystemSimulator20XX("/twenty_eighteen/Day24-ImmuneSystemSimulator20XX-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
